x-pack/plugins/endpoint/public/embeddables/resolver/view/graph_controls.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, ChangeEvent } from 'react';
+import styled from 'styled-components';
+import { EuiRange, EuiPanel, EuiIcon } from '@elastic/eui';
+import { useSelector, useDispatch } from 'react-redux';
+import { ResolverAction, PanDirection } from '../types';
+import * as selectors from '../store/selectors';
+
+/**
+ * Controls for zooming, panning, and centering in Resolver
+ */
+export const GraphControls = styled(
+  React.memo(
+    ({
+      className,
+    }: {
+      /**
+       * A className string provided by `styled`
+       */
+      className?: string;
+    }) => {
+      const dispatch: (action: ResolverAction) => unknown = useDispatch();
+      const scalingFactor = useSelector(selectors.scalingFactor);
+
+      const handleZoomAmountChange = useCallback(
+        (event: ChangeEvent<HTMLInputElement>) => {
+          const valueAsNumber = parseFloat(event.target.value);
+          if (isNaN(valueAsNumber) === false) {
+            dispatch({
+              type: 'userSetZoomLevel',
+              payload: valueAsNumber,
+            });
+          }
+        },
+        [dispatch]
+      );
+
+      const handleCenterClick = useCallback(() => {
+        dispatch({
+          type: 'userSetPositionOfCamera',
+          payload: [0, 0],
+        });
+      }, [dispatch]);
+
+      const handleZoomOutClick = useCallback(() => {
+        dispatch({
+          type: 'userClickedZoomOut',
+        });
+      }, [dispatch]);
+
+      const handleZoomInClick = useCallback(() => {
+        dispatch({
+          type: 'userClickedZoomIn',
+        });
+      }, [dispatch]);
+
+      const handlePanClick = useCallback(
+        (panDirection: PanDirection) => {
+          return () => {
```

Comment:
This will defeat the purpose of `useCallback`, since it creates a new function reference each time it is called, and that function references is passed down. You could skip `useCallback` all together. Or you could use one `useCallback` per each button. or you could put `data-direction="west"` on each button and have one click handler for them all which does `dispatch({ type: 'userClickedPanControl', payload: event.target.dataset.direction })`

Reply:
Changed. I went with skipping `useCallback` all together.